### PR TITLE
ES|QL Add primitive float support to the Compute Engine

### LINKED ISSUE #109178: ESQL: Introduce float-variant for Blocks
### Description

As we're looking at supporting scores, it makes sense to add the float variant for blocks. Separately, as a separate work stream, it might make sense to look at the underlying hashing and grouping structures and create specialized implementations for int (instead of just using long everywhere).

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit adds primitive float support to the Compute Engine - a.k.a. FloatBlock.

ES|QL language and aggregation support can be built atop this engine primitive.  Serialization and wire format compatibility can also be done as a follow up.

closes #109178
relates #109870
relates #109781

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EstimatesRowSize.java [hunk 1]
```diff
@@ -118,6 +118,7 @@ static int estimateSize(DataType dataType) {
             case LONG -> Long.BYTES;
             case NULL -> 0;
             // TODO: provide a specific estimate for aggregated_metrics_double
+            case FLOAT -> throw new EsqlIllegalArgumentException("can't estimate size for float blocks");
```
Comment (nik9000) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EstimatesRowSize.java:121]: This is just `Float.BYTES` I think.

I don't think it'd hurt to add it.

**Review by nik9000 — APPROVED** [review]
Fine by me.

**Comment by nik9000** [comment]
@ChrisHegarty already said it above, but for those playing along I want to make this really explicit:

The ESQL language currently supports `float`, `scaled_float`, and `half_float` style numbers by converting them to `double`. This PR doesn't change that. It *just* makes it possible for the compute engine to consume a `float`. You can't aggregate on it. Or sort it. Or run any functions on it. Or send the blocks over the wire from node to node. Or return them over http. And `float` fields will still be loaded as `double` in ESQL.

Those things can come in time. Carefully. We're not currently sure if it's a breaking change for those fields to suddenly get loaded as smaller types. A lot of what I said above is "just write more code". But the backwards compatibility stuff is quite interesting.

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**